src/ssr-template-helper.js

```diff
@@ -58,65 +60,66 @@ export class SsrTemplateHelper {
   }
 
   /**
-   * Proxies xhr and template rendering to the viewer and renders
-   * the response.
+   * Proxies xhr and template rendering to the viewer and renders the response.
    * @param {!Element} element
+   * @param {!./service/xhr-impl.FetchData} fetchData The fetch/XHR related data.
+   * @param {?SsrTemplateDef=} opt_templates Response templates to pass into
+   *     the payload. If provided, finding the template in the passed in
+   *     element is not attempted.
+   * @param {!Object=} opt_attributes Additional JSON to send to viewer.
    * return {!Promise<{data:{?JsonObject|string|undefined}}>}
    */
-  fetchAndRenderTemplate(element) {
-    const inputsAsJson = this.getElementInputsAsJson_(element);
-    const elementAttrsAsJson =
-        this.getElementAttributesAsJson_(element);
-    elementAttrsAsJson['inputData'] = inputsAsJson;
-    const template = this.templates_.maybeFindTemplate(element);
+  fetchAndRenderTemplate(
+    element, fetchData, opt_templates = null, opt_attributes = {}) {
     let mustacheTemplate;
-    if (template) {
-      // The document fragment can't be used in the message channel API thus
-      // serializeToString for a string representation of the dom tree.
-      mustacheTemplate = this.xmls_.serializeToString(
-          this.templates_.findTemplate(element));
+    if (!opt_templates) {
+      const template = this.templates_.maybeFindTemplate(element);
+      if (template) {
+        // The document fragment can't be used in the message channel API thus
+        // serializeToString for a string representation of the dom tree.
+        mustacheTemplate = this.xmls_.serializeToString(
```

Comment:
Don't we need to do this to the `opt_templates`?

Reply:
yes, i do so in the buildPayload_ method.